geo/src/algorithm/contains/rect.rs

```diff
@@ -0,0 +1,40 @@
+use super::Contains;
+use crate::*;
+
+// ┌──────────────────────────┐
+// │ Implementations for Rect │
+// └──────────────────────────┘
+
+impl<T> Contains<Coordinate<T>> for Rect<T>
+where
+    T: CoordinateType,
+{
+    fn contains(&self, coord: &Coordinate<T>) -> bool {
+        coord.x > self.min().x
+            && coord.x < self.max().x
+            && coord.y > self.min().y
+            && coord.y < self.max().y
+    }
+}
+
+impl<T> Contains<Point<T>> for Rect<T>
+where
+    T: CoordinateType,
+{
+    fn contains(&self, p: &Point<T>) -> bool {
+        self.contains(&p.0)
+    }
+}
+
+impl<T> Contains<Rect<T>> for Rect<T>
+where
+    T: CoordinateType,
+{
+    fn contains(&self, bounding_rect: &Rect<T>) -> bool {
+        // All points of LineString must be in the polygon ?
+        self.min().x <= bounding_rect.min().x
```

Comment:
So a degenerate Rect which is either a Line or a Point will be `contained` if it's on the boundary - I don't think that's correct, is it?

Reply:
Good catch!  The previous implementation was the same with strict checking, which would also miss some edge cases.  Want to address these in a different PR (needs a few extra impls, utils).   Let me add a todo comment here so we get back to it.